integration/tests/epochs/epoch_test.go

```diff
@@ -142,68 +137,40 @@ func (s *Suite) TestEpochJoin() {
 
 	role := flow.RoleAccess
 
-	// stake a new node
-	info := s.StakeNode(ctx, env, role)
+	// stake a new node and start the docker container
+	info, snapshot, _ := s.StakeAndStartNewNode(ctx, env, role)
 
-	// ensure node ID in approved list
-	approvedNodes := s.ExecuteReadApprovedNodesScript(ctx, env)
-	require.Contains(s.T(), approvedNodes.(cadence.Array).Values, cadence.String(info.NodeID.String()), fmt.Sprintf("expected new node to be in approved nodes list: %x", info.NodeID))
+	// make sure the network is healthy after adding new AN
+	s.assertNetworkHealthyAfterANChange(ctx, env, snapshot, info)
 
-	// check if node is in proposed table
-	proposedTable := s.ExecuteGetProposedTableScript(ctx, env, info.NodeID)
-	require.Contains(s.T(), proposedTable.(cadence.Array).Values, cadence.String(info.NodeID.String()), fmt.Sprintf("expected new node to be in proposed table: %x", info.NodeID))
-
-	containerConfigs := []func(config *testnet.NodeConfig){
-		testnet.WithLogLevel(zerolog.DebugLevel),
-		testnet.WithID(info.NodeID),
-	}
-
-	nodeConfig := testnet.NewNodeConfig(role, containerConfigs...)
-	testContainerConfig := testnet.NewContainerConfig(info.ContainerName, nodeConfig, info.NetworkingKey, info.StakingKey)
-	err := testContainerConfig.WriteKeyFiles(s.net.BootstrapDir, flow.Localnet, info.MachineAccountAddress, encodable.MachineAccountPrivKey{PrivateKey: info.MachineAccountKey}, role)
-	require.NoError(s.T(), err)
-
-	s.WaitForPhase(ctx, flow.EpochPhaseSetup)
-	snapshot, err := s.client.GetLatestProtocolSnapshot(ctx)
-	require.NoError(s.T(), err)
-
-	//add our container to the network
-	err = s.net.AddNode(s.T(), s.net.BootstrapDir, testContainerConfig)
-	require.NoError(s.T(), err, "failed to add container to network")
+	s.net.StopContainers()
+}
 
-	//start our test container
-	testContainer := s.net.ContainerByID(info.NodeID)
-	testContainer.WriteRootSnapshot(snapshot)
-	testContainer.Container.Start(ctx)
+// TestEpochLeave this test add a new node to the network and after some time un-stake and remove a node
+// from the network of the same node type. The test then makes the assertions necessary for the network
+// to be considered healthy
+func (s *Suite) TestEpochLeave() {
```

Comment:
I think this test practically has full overlap with the current Access-node-based `TestEpochJoin` test. Since these tests will be so expensive in terms of CI time, in the cases where the scope of "leave" fully encompasses the scope of "join" we should just keep one (and be sure to document how both cases remain covered).

Reply:
https://github.com/onflow/flow-go/pull/1728/commits/657e37d5bb69949bc8b30a6ca1acf70570779c71